File: ut/python/myclient.py
import sys
sys.path.append("../gen-py")

import agnos 
from agnos.servers import SimpleServer
import RemoteFiles


c = RemoteFiles.Client.connect("localhost", 17731)
f = c.open("/tmp/foo", "w")
print f
f.write("hello world")
s = f.stat()
print s
try:
    f.read(10)
except RemoteFiles.UnderlyingIOError, ex:
    print ex
f.close()
try:
    f.write("dlrow olleh")
except agnos.GenericError, ex:
    print ex

f1 = c.open("/tmp/foo", "r")
f2 = c.open("/tmp/foo2", "w")
c.copy(f1, f2)
f1.close()
f2.close()

f2 = c.open("/tmp/foo2", "r")
data = f2.read(100)
print "copy = %r" % (data,)

File: ut/java/myclient.java
import java.util.*;
import java.io.*;
import RemoteFiles.Types.*;
import RemoteFiles.Service.*;

public class myclient
{
	public static void main(String[] args)
	{
		try {
			Client c = new Client(
					new agnos.Servers.SocketTransport("localhost", 17732));
			
			try {
				FileProxy f = c.open("/tmp/foo", "w");
				System.out.println("f = " + f);
				f.write("hello world".getBytes());

				System.out.println("filename = " + f.get_filename());
				System.out.println("stat = " + f.stat());
				
				boolean got_exc = false;
				try {
					f.flush();
				} catch (UnderlyingIOError exc) {
					got_exc = true;
					System.out.println("matched: " + exc);
				}
				assert(got_exc);
				
				try {
					// should cause NullPointer, since opened for writing
					f.read(10); 
				}
				catch (agnos.Protocol.GenericError exc) {
					System.out.println("matched: " + exc);
				}
				
				f.close();

				FileProxy f1 = c.open("/tmp/foo", "r");
				FileProxy f2 = c.open("/tmp/foo2", "w");
				c.copy(f1, f2);
				f1.close();
				f2.close();

				f2 = c.open("/tmp/foo2", "r");
				byte[] data = f2.read(100);
				System.out.println("copy = " + new String(data));
				
				System.out.println("client finished successfully");

			} finally {
				c.close();
			}
		} catch (Exception ex) {
			ex.printStackTrace();
		}
	}
}


